limiter.go

```diff
@@ -86,6 +87,9 @@ func (r *rateLimiter) Status(key string) (bool, float64, error) {
 
 func (l *rateLimiter) Handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		l.mu.Lock()
```

Comment:
are you sure we need this...? can you provide a test case showing the case is necessary...?

the LimitCounter methods have their own locks within their implementation

Reply:
I'm sure. It works ok with standard limiter and doesn't work with httprate redis module (test is simple just send a lot of requests and you will see you can send more than limit). Probably it is due to bigger redis overhead in comparison to stadard limiter.